### PR TITLE
[#22] Add couchdb helper methods

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: common/util/retry/retry.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
```
Comment (troyronda) [inline common/util/retry/retry.go]: missing test file

File: core/ledger/util/couchdb/couchdb_ext.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hyperledger/fabric/common/util/retry"
+	"github.com/pkg/errors"
+)
+
+// CreateNewIndexWithRetry method provides a function creating an index but retries on failure
```
Comment (troyronda) [inline core/ledger/util/couchdb/couchdb_ext.go]: CreateNew :)

File: common/util/retry/retry.go (hunk 1 above)
Comment (fqutishat) [reply common/util/retry/retry.go]: done

**Comment by fqutishat** [comment]
Verify

**Pull request merged**